code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+
+#Need to ensure there is a total for standard delays (HSC/PFR added together). This is due to the format of the publication data sheet.
+#datafile22a<-filter(datafile21,reason_grp_high_level!="ALL") 
+#Rename reason_grp_high_level as Standard where there isn't a Code 9
+datafile22<-datafile21%>% mutate(reas1=
+                                   if_else(reas1%in%c("HEALTH AND SOCIAL CARE REASONS","PATIENT/CARER/FAMILY-RELATED REASONS"),"STANDARD",reas1))
+table(datafile22$reas1)
+
+#select standard only.
+datafile23<-filter(datafile22,reas1=="STANDARD")
+# aggregate 
+
+datafile24<- datafile23 %>% 
+  group_by(Healthboard,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+datafile24<-datafile24%>%rename(obds3=OBDs)
+
+#match files
+datafile25 <- left_join(datafile24,datafile21,
+                        by = c(("Healthboard" = "Healthboard"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+datafile25a<-bind_rows(datafile24,datafile21)
+
+
+#replace obds in correct column for all reason groups
+datafile26<-datafile25a %>% mutate(OBDs=
+                                     if_else(!is.na(obds3),obds3,OBDs))
+
+#remove surplus columns
+datafile26<-select(datafile26,-obds3,-obds2)
+
+#match back to template to ensure every row is populated
+
+datafile27 <- left_join(hbbedstemplate,datafile26,
+                        by = c(("Healthboard" = "Healthboard"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+#replace obds in correct column for all reason groups
+datafile27<-datafile27 %>% mutate(OBDs=
+                                    if_else(is.na(OBDs),0L,OBDs))
+
+#remove surplus columns
+datafile27<-select(datafile27,-obds2)
+
+datafile27<-arrange(datafile27,datafile27$Healthboard,datafile27$age_grp,datafile27$reas1)
+
+
+### 17. Save out bed days HB file ----
+
+write.xlsx(datafile27,paste0(filepath,"HB bed days data sheet_R.xlsx"))
+
+#########################################
+### 18. LA bed days  ----
+#########################################
+labedstemplate<-read.csv(paste0(filepath2,"la_template.csv"))
+
+labedstemplate <- labedstemplate %>% 
+  rename(LocalAuthorityArea = LA,
+         age_grp = age)
+
+labedstemplate<-labedstemplate%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+datafile20<-datafile20%>% mutate(LocalAuthorityArea=toupper(LocalAuthorityArea),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+##################################################################
+
+#amend variables as necessary in order for matchi_numberng to work
+
+labeddaysdatasheetminusstandard <- left_join(datafile20,labedstemplate,
+                                             by = c(("LocalAuthorityArea" = "LocalAuthorityArea"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(labeddaysdatasheetminusstandard,LocalAuthorityArea,age_grp,reas1) # arrange data in order for tables
```

Comment:
The `left_join()` function needs to have the common variables listed. As they are the same name in both of these data frames, they don't need to be listed to equal each other.
The `arrange()` function needs a data frame input as the first argument, or if using the pipe, to be part of that chain. As it stands, `arrange()` isn't doing anything here. Let's attach it to the previous code.
```suggestion
labeddaysdatasheetminusstandard <- left_join(datafile20, labedstemplate, 
    by = c("LocalAuthorityArea", "age_grp", "reas1")) %>% 
    arrange(labeddaysdatasheetminusstandard, LocalAuthorityArea, age_grp, reas1)
```